scopes/lanes/ui/navigation/lane-switcher/lane-switcher.tsx

```diff
@@ -14,13 +14,20 @@ export type LaneSwitcherProps = {
 
 export function LaneSwitcher({ className, groupByScope = true, ...rest }: LaneSwitcherProps) {
   const { lanesModel } = useLanes();
+  const [viewedLane, setViewedLane] = useState(lanesModel?.viewedLane);
+
+  useEffect(() => {
+    if (viewedLane?.id && lanesModel?.viewedLane?.id.isEqual(viewedLane?.id)) return undefined;
+    setViewedLane(lanesModel?.viewedLane);
+    return undefined;
+  }, [lanesModel?.viewedLane?.id.toString()]);
```

Comment:
Why not check if the condition is false and then run the set?
and remove the two return of undefined